src/v/storage/segment_index.h

```diff
@@ -49,7 +50,18 @@ class segment_index {
     static constexpr size_t default_data_buffer_step = 4096 * 8;
 
     segment_index(
-      ss::sstring filename, ss::file, model::offset base, size_t step);
+      ss::sstring filename,
+      model::offset base,
+      size_t step,
+      debug_sanitize_files);
+
+    /** Constructor with mock file content for unit testing */
+    segment_index(
```

Comment:
i liked that trick deployed elsewhere of making this private and making the test fixture a friend?

Reply:
Added a commit to clean this up into a private constructor and update the unit tests.